zeebe/engine/src/main/java/io/camunda/zeebe/engine/processing/processinstance/ProcessInstanceMigrationMigrateProcessor.java

```diff
@@ -491,6 +493,19 @@ private static Set<ExecutableSequenceFlow> getSequenceFlowsToMigrate(
         .collect(Collectors.toSet());
   }
 
+  private static void checkSequenceFlowIdsIntegrity(
+      final ElementInstance elementInstance, final long processInstanceKey) {
+    if (elementInstance.getActiveSequenceFlows() != elementInstance.getActiveSequenceFlowsCount()) {
+      throw new SafetyCheckFailedException(
```

Comment:
~❌ The safety check exception leads to a logged error (which we'll see in Google Error Reporting if it occurs on SaaS). However, this is an expected case, not an error. Let's handle it the same as the concurrent case.~

EDIT: no longer needed